pkg/auth/oauth2authenticator.go

```diff
@@ -136,15 +120,45 @@ func createVerifier(count int) []byte {
 	lut := []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~")
 	verifier := make([]byte, count)
 
-	// TODO is this good enough?
 	for i := range verifier {
-		index := rand.Int() % len(lut)
+		index := randIndex(len(lut))
 		verifier[i] = lut[index]
 	}
 
 	return verifier
 }
 
+const maxRandIndexLimit = 256
+
+// randIndex provides a secure random number in the range [0, limit).
+//
+// Rejection sampling and modulo reduction are applied to avoid modulo bias and
+// ensure a uniform distribution.
+//
+// This function panics if the limit exceeds maxRandIndexLimit, or if a secure
+// random source cannot be read.
+//
+// Further reading:
+// https://research.kudelskisecurity.com/2020/07/28/the-definitive-guide-to-modulo-bias-and-how-to-avoid-it/
+// https://stackoverflow.com/a/10984975
+func randIndex(limit int) int {
+	if limit > maxRandIndexLimit {
+		panic(fmt.Sprintf("unsupported randIndex limit %d", limit))
+	}
+	rejectLimit := 256 - (256 % limit)
```

Comment:
Question: I'm not sure yet, why it wouldn't be enough to do only rejection sampling, using the given limit.

Reply:
We could, but we'd end up rejecting a lot of potentially useful values for a small `limit`.

Rejecting any sample greater than the largest multiple of `limit` makes more use of the raw random value. The selected sample `% limit` will still have a uniform distribution.

Reply:
Actually.. I'm feeling like it's probably safer to use `crypto/rand.Int` here. For the ranges we're working with... it's not too different from what I was attempting here. Best to keep it simple.